extensions/amp-connatix-player/0.1/amp-connatix-player.js

```diff
@@ -323,8 +330,33 @@ export class AmpConnatixPlayer extends AMP.BaseElement {
     // bind to amp consent and send consent info to the iframe content and propagate to player
     this.bindToAmpConsent_();
 
+    const mutationObserverConfig = {
+      attributes: true,
+      childList: false,
+      subtree: false,
+    };
+
+    const callback = (mutationList) => {
+      for (const mutation of mutationList) {
+        if (
+          mutation.type === 'attributes' &&
+          mutation.attributeName === 'class'
+        ) {
+          this.sendCommand_(
+            mutation.target.classList.contains('i-amphtml-video-docked')
+              ? 'dock'
+              : 'undock'
+          );
+        }
+      }
+    };
+
+    const observer = new MutationObserver(callback);
```

Comment:
Can you move the observer to a member property. When the player is being destroyed, we can clean up the observer.